google-cloud-video-live_stream/samples/acceptance/helper.rb

```diff
@@ -19,33 +19,141 @@
 require "google/cloud/video/live_stream"
 
 require_relative "../../../.toys/.lib/sample_loader"
+require_relative "channel_definition"
+require_relative "input_definition"
+
+DELETION_THRESHOLD_TIME_HOURS_IN_SECONDS = 10_800
 
 class LiveStreamSnippetSpec < Minitest::Spec
   let(:client) { Google::Cloud::Video::LiveStream.livestream_service }
 
   let(:project_id) { ENV["GOOGLE_CLOUD_PROJECT"] || raise("missing GOOGLE_CLOUD_PROJECT") }
-
   let(:location_id) { "us-central1" }
+  let(:location_path) { client.location_path project: project_id, location: location_id }
+
   let(:input_id) { "my-input-test-#{(Time.now.to_f * 1000).to_i}" }
   let(:input_name) { "projects/#{project_id}/locations/#{location_id}/inputs/#{input_id}" }
-  let(:input_location_path) { client.location_path project: project_id, location: location_id }
+  let(:input_path) { client.input_path project: project_id, location: location, input: input_id }
+
+  let(:update_input_id) { "my-update-input-test-#{(Time.now.to_f * 1000).to_i}" }
+  let(:update_input_name) { "projects/#{project_id}/locations/#{location_id}/inputs/#{update_input_id}" }
+  let(:update_input_path) { client.input_path project: project_id, location: location, input: update_input_id }
+
+  let(:channel_id) { "my-channel-#{(Time.now.to_f * 1000).to_i}" }
+  let(:channel_name) { "projects/#{project_id}/locations/#{location_id}/channels/#{channel_id}" }
+  let(:output_uri) { "gs://my-bucket/my-output-folder/" }
+
+  before do
+    @channel_created_started = false
+    @channel_created_stopped = false
+    @input_created = false
+    @update_input_created = false
+    # Remove old channels and inputs in the test project if they exist
+    response = client.list_channels parent: location_path
+    response.each do |channel|
+      tmp = channel.name.to_s.split "-"
+      create_time = tmp.last.to_i
+      now = (Time.now.to_f * 1000).to_i
+      next if create_time >= (now - DELETION_THRESHOLD_TIME_HOURS_IN_SECONDS)
```

Comment:
It looks like both `now` and `create_time` are in milliseconds (i.e. `Time.now.to_f` returns times in seconds, and you multiply by 1000). So you probably want `DELETION_THRESHOLD_TIME_HOURS_IN_SECONDS` to be a factor of 1000 larger too. Currently it's effectively set to 10 seconds which might be short enough to cause test flakes if multiple copies of the test are running at the same time.

Reply:
Changed DELETION_THRESHOLD_TIME_HOURS_IN_SECONDS name and multiplied by 1000